src/anyio/_backends/_asyncio.py

```diff
@@ -499,28 +499,25 @@ async def __aexit__(
         while self.cancel_scope._tasks:
             try:
                 await asyncio.wait(self.cancel_scope._tasks)
-            except asyncio.CancelledError:
+            except asyncio.CancelledError as exc:
+                # This task was cancelled natively; reraise the CancelledError later
+                # unless this task was already interrupted by another exception
                 self.cancel_scope.cancel()
+                if exc_val is None:
+                    exc_val = exc
+                    ignore_exception = False
+                    self._exceptions.append(exc)
```

Comment:
Why do you push the `CancelledError` onto `self._exceptions` if it will be dropped on the `group.split` call?

Reply:
Because without that, the `if self._exceptions` would be `False`, so we never end up re-raising the `CancelledError` unless other exceptions crop up too.

Reply:
Ah I missed that conditional — just outside the diff :) thanks!